### PR TITLE
When properties appear in propstat with no value, don't call deserializers

### PR DESCRIPTION
**Pull request by evert**
Related to #707 #709 

@DominikTo

### CODE REVIEW

File: tests/Sabre/DAV/Xml/Element/ResponseTest.php [hunk 1]
```diff
@@ -179,4 +179,71 @@ function testSerializeBreak() {
...
+    </d:prop>
+    <d:status>HTTP/1.1 404 Not Found</d:status>
+  </d:propstat>
+</d:response>
+';
+
+        $result = $this->parse($xml, [
+            '{DAV:}response' => 'Sabre\DAV\Xml\Element\Response',
+            '{DAV:}foo' => function($reader) {
+                throw new \LogicException('This should never happned');
```
Comment (DominikTo) [inline tests/Sabre/DAV/Xml/Element/ResponseTest.php]: Either "never have happened" or "never happen". ;-)

**Comment by DominikTo** [comment]
:+1:

**Comment by Hywan** [comment]
:+1:

File: lib/DAV/Xml/Element/Response.php [hunk 2]
```diff
@@ -161,7 +161,46 @@ function xmlSerialize(Writer $writer) {
...
+        $reader->pushContext();
+
+        $reader->elementMap['{DAV:}propstat'] = 'Sabre\\Xml\\Element\\KeyValue';
+
+        // We are overriding the parser for {DAV:}prop. This deserializer is
+        // almost identical to the one for Sabre\Xml\Element\KeyValue.
+        //
+        // The difference is that if there are any child-elements inside of
+        // {DAV:}prop, that have no value, normally any deserializers are
+        // called. But we don't want this, because a singlular element without
```
Comment (Hywan) [inline lib/DAV/Xml/Element/Response.php]:
```
- singlular
+ singular
```

**Pull request merged**